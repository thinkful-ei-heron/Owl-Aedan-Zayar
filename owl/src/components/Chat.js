import React from 'react';
// import './chat.css';

export default class Chat extends React.Component {
  renderChatEvent(event, index) {
    let name = this.props.participants.find(item => item.id === event.participantId).name;
    let content = '';
    switch (event.type) {
      case 'thumbs-up':
        content += 'gave a thumbs up';
        break;
      case 'thumbs-down':
        content += 'gave a thumbs down';
        break;
      case 'raise-hand':
        content += 'raised their hand';
        break;
      case 'clap':
        content += 'clapped';
        break;
      case 'join':
        content += 'joined.';
        break;
      case 'leave':
        content += 'left';
        break;
      case 'join-stage':
        content += 'joined the stage';
        break;
      case 'leave-stage':
        content += 'left the stage';
        break;
    } 
    return (
      <li key={index} className="chatEvent">
        <span className="chat-event-name">{name} </span>
        <span className="chat-event-content">{content}</span>
      </li>
    )
  }


  renderMessageEvent(event, index) {
    let message = event.message;
    let time = new Date(event.time).toTimeString();
    let sender = this.props.participants.find(item => item.id === event.participantId);
    let name = sender.name;
    let avatar = sender.avatar;

    return (
      <li key={index} className="messageEvent">
        <div>
          <img src={avatar} alt="avatar" />
          <span>{name}</span>
          <time>{time}</time>
        </div>
        <div>
          <p>{message}</p>
        </div>
      </li>
    );
  };
  render() {

    let events = this.props.events;
    let chatEvents = [];

    events.forEach((event, index) => {

      if (event.type === 'message') {
        chatEvents.push(this.renderMessageEvent(event, index));
      } else {
        chatEvents.push(this.renderChatEvent(event, index));
      }
    })

    return (
      <div className="Chat">
        <ul>
          {chatEvents}
        </ul>
        
      </div>
    );
  }
}