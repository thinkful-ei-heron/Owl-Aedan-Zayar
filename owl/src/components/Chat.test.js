import ReactDOM from 'react-dom';
import React from 'react';
import renderer from 'react-test-renderer';
import Chat from './Chat';
import store from '../store.js';

describe('Chat component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <Chat participants={store.participants} events={store.chatEvents} />, div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it('Renders as expected', () => {
    const tree = renderer
      .create(<Chat participants={store.participants} events={store.chatEvents}         />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});