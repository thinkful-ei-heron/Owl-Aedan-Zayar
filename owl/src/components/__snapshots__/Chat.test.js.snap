// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chat component Renders as expected 1`] = `
<div
  className="chat"
>
  <ul>
    <li
      className="messageEvent"
    >
      <div>
        <img
          alt="avatar"
          src="https://robohash.org/utlaborumfugit.jpg?size=200x200&set=set1"
        />
        <span>
          Axe Kubicka
        </span>
        <time>
          07:50:46 GMT-0500 (EST)
        </time>
      </div>
      <div>
        <p>
          Hello world
        </p>
      </div>
    </li>
    <li
      className="chatEvent"
    >
      <span
        className="chat-event-name"
      >
        Caty Flucker
         
      </span>
      <span
        className="chat-event-content"
      >
        gave a thumbs up
      </span>
    </li>
    <li
      className="chatEvent"
    >
      <span
        className="chat-event-name"
      >
        Frank Runciman
         
      </span>
      <span
        className="chat-event-content"
      >
        gave a thumbs down
      </span>
    </li>
    <li
      className="chatEvent"
    >
      <span
        className="chat-event-name"
      >
        Koren Templeton
         
      </span>
      <span
        className="chat-event-content"
      >
        raised their hand
      </span>
    </li>
    <li
      className="chatEvent"
    >
      <span
        className="chat-event-name"
      >
        Koren Templeton
         
      </span>
      <span
        className="chat-event-content"
      >
        clapped
      </span>
    </li>
    <li
      className="chatEvent"
    >
      <span
        className="chat-event-name"
      >
        Koren Templeton
         
      </span>
      <span
        className="chat-event-content"
      >
        joined.
      </span>
    </li>
    <li
      className="chatEvent"
    >
      <span
        className="chat-event-name"
      >
        Ashla Attwool
         
      </span>
      <span
        className="chat-event-content"
      >
        left
      </span>
    </li>
    <li
      className="chatEvent"
    >
      <span
        className="chat-event-name"
      >
        Axe Kubicka
         
      </span>
      <span
        className="chat-event-content"
      >
        joined the stage
      </span>
    </li>
    <li
      className="chatEvent"
    >
      <span
        className="chat-event-name"
      >
        Axe Kubicka
         
      </span>
      <span
        className="chat-event-content"
      >
        left the stage
      </span>
    </li>
  </ul>
</div>
`;
